utils/open-control-utils.js

```diff
@@ -76,53 +73,180 @@ var caseHyphenInsensitive = function(key){
+                    else{
+                        return r.json()
+                    }
+                })
+                .then((doc)=>{
+                    formatComponent(doc, standardsCompliance);    
+                    return callback(arr, i,callback,standardsCompliance);
+                    
+                })
+            })
+        
         })
+        .then(standardsCompliance=>{
+            return certifications.forEachConsecutively(function(arr,i,callback){
+                let _ = arr[i];
+                if(_.satisfied===undefined){
+                    _ = Object.assign(_,{
+                        controlFamilies: 0, totalControls: 0, inheritedCompliance: 0, proceduralControls: 0, technicalControls: 0,
+
+                        satisfied: 0,
+                        partial: 0,
+                        noncompliant: 0
+                    })
+                }
+                return fetch(_.url).then(r=>{
+                    if(r.url.indexOf('undefined') >=0) return null;
+                    else{
+                        return r.json()
+                    }
+                }).then((doc)=>{
+                    if(doc !== null)
+                    for(let standardKey in doc.standards){
+                        if(standardsCompliance[standardKey]===undefined) break;
+                        for(let controlKey in doc.standards[standardKey]){
+                            _.totalControls++;
+                            if(standardsCompliance[standardKey][controlKey]==undefined)continue;
+                            switch(standardsCompliance[standardKey][controlKey].implementation_status){
+                                case 'complete':
+                                _.satisfied++;
+                                break;
+                                case 'patrial':
+                                _.partial++;
+                                break;
+                                default:
+                                _.noncompliant++;
+                                break;
+                            }
+                        }
+                    }
+                    return callback(arr, i+1,callback,certifications);
+                })
+        });
     })
+    
+}
+
+module.exports.getComponents = function(callback){
+    fetch(constants.get_components_url).then(r => r.json())
+      .then(data => {
+        // localStorage.setItem('users',JSON.stringify(data));
+        callback(data);
+      })
+      .catch(e => console.log(e));
 }
+
+module.exports.getComponent = function(url,callback){
+    fetch(url).then(r=>r.json())
+    .then(data=>{
+        //format data
+        let standardsCompliance={};
+        formatComponent(data, standardsCompliance);
+        let meta = {
+            satisfied: 0,
+            partial: 0,
+            noncompliant: 0,
+            totalControls:0
+        };
+        for(let standardKey in standardsCompliance){
```

Comment:
```suggestion
        // Assume content has been though validation, no need for
        // validating implementation_status (e.g. Complete vs complete vs COMPLETE)
        for(let standardKey in standardsCompliance){
```

Reply:
Yea I'll open new PRs and fix those. (Thanks for the review!